tests/storage/datavolume.go

```diff
@@ -759,6 +759,33 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 				// Wait for the owner reference to disappear
 				Eventually(ThisDVWith(vm.Namespace, vm.Spec.DataVolumeTemplates[0].Name), 100).Should(Not(BeOwned()))
 			})
+
+			It("[test_id:XXXX]should fail creating a VMI with ExceededQuota status and Failure condition when using ResourceQuota storage limits", func() {
```

Comment:
Can we transform this into a unit test instead of a functional one?

Reply:
The test actually checks a `ResourceQuota` failing a `PVC` creation in its validating webhook, which leads to `DV` conditions with `ErrExceededQuota` and then to the `VM` `Failure` condition with the detailed message. Since it fails early by the `PVC` webhook so no `PVC` or `VMI` are created, it is quite cheap, and not sure how wise will it be to transform it to a unit test checking just the DV condition -> VM condition part.

Reply:
Sure.